### ISSUE TITLE
自己训练的模型，无法test，在用continue_train时出现问题

### ISSUE THREAD

**Issue by GuoJingWeng** [KumapowerLIU/Rethinking-Inpainting-MEDFE#12]
test时，对于模型的加载出现问题
![image](https://user-images.githubusercontent.com/40395056/98239596-27098e00-1fa3-11eb-9061-d4e2eba02230.png)

continue_train时，在MEDFN.py中没有引用load_network出现error
![image](https://user-images.githubusercontent.com/40395056/98239549-16591800-1fa3-11eb-86d7-d5736077c8eb.png)

**Comment by Xiefan-Guo** [comment]
Maybe you can modify the code as follows, it works for me.
```
model.netEN.module.load_state_dict(torch.load("EN.pth")['net'])
model.netDE.module.load_state_dict(torch.load("DE.pth")['net'])
model.netMEDFE.module.load_state_dict(torch.load("MEDFE.pth")['net'])
```

**Comment by KumapowerLIU** [comment]
I change the code and it is correct now

**Comment by GuoJingWeng** [comment]
> Maybe you can modify the code as follows, it works for me.
> 
> ```
> model.netEN.module.load_state_dict(torch.load("EN.pth")['net'])
> model.netDE.module.load_state_dict(torch.load("DE.pth")['net'])
> model.netMEDFE.module.load_state_dict(torch.load("MEDFE.pth")['net'])
> ```

Thank you very much. It's working properly

**Comment by GuoJingWeng** [comment]
> 我更改了代码，现在是正确的
Thank you very much. Your perfect work has greatly promoted my work

**Comment by ZHANG-hengzhi** [comment]
您好，我最近也在复现这个代码，但是遇到一些问题，可以和您交流一下吗

**Comment by ZHANG-hengzhi** [comment]
您好，我更改了opt里面的参数，但是为什么每次加载opt都会恢复最初始的值

**Comment by ZHANG-hengzhi** [comment]
我在训练自己的代码时会报错AttributeError: 'Tensor' object has no attribute 'bool'
![image](https://user-images.githubusercontent.com/63292940/117535868-65ebcf80-b02a-11eb-9143-9494e377b00c.png)
请问是什么原因啊，非常感谢

**Comment by fxcdl** [comment]
请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了

**Comment by GuoJingWeng** [comment]
他只是方便了数据加载，其实你看那部分源码就是只用了mask和原始图像的数据



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2021年7月21日(周三) 下午3:20
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)




 
请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了

**Comment by fxcdl** [comment]
请问为什么还要用原始图像，直接输入MASK不就应该能得到去遮挡的图吗




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "KumapowerLIU/Rethinking-Inpainting-MEDFE"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月21日(星期三) 下午3:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)





 他只是方便了数据加载，其实你看那部分源码就是只用了mask和原始图像的数据 
 
 
 
 ---原始邮件--- 
 发件人: ***@***.***&amp;gt; 
 发送时间: 2021年7月21日(周三) 下午3:20 
 收件人: ***@***.***&amp;gt;; 
 抄送: ***@***.******@***.***&amp;gt;; 
 主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 
 
 
 
 
   
 请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了

**Comment by GuoJingWeng** [comment]
我说的原始图像就是待修复图像，mask是黑白图像，白色区域是标注的区域



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2021年7月21日(周三) 下午3:32
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)




 请问为什么还要用原始图像，直接输入MASK不就应该能得到去遮挡的图吗 
 
 
 
 
 ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 
 发件人:                                                                                                                        "KumapowerLIU/Rethinking-Inpainting-MEDFE"                                                                                    ***@***.***&amp;gt;; 
 发送时间:&amp;nbsp;2021年7月21日(星期三) 下午3:30 
 ***@***.***&amp;gt;; 
 ***@***.******@***.***&amp;gt;; 
 主题:&amp;nbsp;Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 
 
 
 
 
 
  他只是方便了数据加载，其实你看那部分源码就是只用了mask和原始图像的数据  
   
   
   
  ---原始邮件---  
  发件人: ***@***.***&amp;amp;gt;  
  发送时间: 2021年7月21日(周三) 下午3:20  
  收件人: ***@***.***&amp;amp;gt;;  
  抄送: ***@***.******@***.***&amp;amp;gt;;  
  主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)  
   
   
   
   
     
  请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了

**Comment by fxcdl** [comment]
明白了，是我弄错了，万分感谢！！！




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "KumapowerLIU/Rethinking-Inpainting-MEDFE"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月21日(星期三) 下午3:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)





 我说的原始图像就是待修复图像，mask是黑白图像，白色区域是标注的区域 
 
 
 
 ---原始邮件--- 
 发件人: ***@***.***&amp;gt; 
 发送时间: 2021年7月21日(周三) 下午3:32 
 收件人: ***@***.***&amp;gt;; 
 抄送: ***@***.******@***.***&amp;gt;; 
 主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 
 
 
 
 
  请问为什么还要用原始图像，直接输入MASK不就应该能得到去遮挡的图吗  
   
   
   
   
  ------------------&amp;amp;nbsp;原始邮件&amp;amp;nbsp;------------------  
  发件人:                                                                                                                        "KumapowerLIU/Rethinking-Inpainting-MEDFE"                                                                                    ***@***.***&amp;amp;gt;;  
  发送时间:&amp;amp;nbsp;2021年7月21日(星期三) 下午3:30  
  ***@***.***&amp;amp;gt;;  
  ***@***.******@***.***&amp;amp;gt;;  
  主题:&amp;amp;nbsp;Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)  
   
   
   
   
   
   他只是方便了数据加载，其实你看那部分源码就是只用了mask和原始图像的数据   
     
     
     
   ---原始邮件---   
   发件人: ***@***.***&amp;amp;amp;gt;   
   发送时间: 2021年7月21日(周三) 下午3:20   
   收件人: ***@***.***&amp;amp;amp;gt;;   
   抄送: ***@***.******@***.***&amp;amp;amp;gt;;   
   主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12)   
     
     
     
     
       
   请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了

**Comment by lh-13** [comment]
> 我说的原始图像就是待修复图像，mask是黑白图像，白色区域是标注的区域
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2021年7月21日(周三) 下午3:32 收件人: ***@***.***&gt;; 抄送: ***@***.******@***.***&gt;; 主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 请问为什么还要用原始图像，直接输入MASK不就应该能得到去遮挡的图吗
> ------------------&amp;nbsp;原始邮件&amp;nbsp;------------------ 发件人: "KumapowerLIU/Rethinking-Inpainting-MEDFE" ***@***.***&amp;gt;; 发送时间:&amp;nbsp;2021年7月21日(星期三) 下午3:30 ***@***.***&amp;gt;; ***@***.******@***.***&amp;gt;; 主题:&amp;nbsp;Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 他只是方便了数据加载，其实你看那部分源码就是只用了mask和原始图像的数据
> ---原始邮件--- 发件人: ***@***.***&amp;amp;gt; 发送时间: 2021年7月21日(周三) 下午3:20 收件人: ***@***.***&amp;amp;gt;; 抄送: ***@***.******@***.***&amp;amp;gt;; 主题: Re: [KumapowerLIU/Rethinking-Inpainting-MEDFE] 自己训练的模型，无法test，在用continue_train时出现问题 (#12) 请问测试的时候不是应该只需要MASK吗，为什么test,py里三种图都用了 — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.
@GuoJingWeng @fxcdl 你们好，我也对这里输出结果的获取有些疑问，在test.py里还是用到了detail图像，也就是gt图像：
fake_out = fake_out.detach().cpu() * mask + detail*(1-mask)
这里要怎么解释？

**Comment by lh-13** [comment]
@GuoJingWeng @fxcdl 你们好，我也对这里输出结果的获取有些疑问，在test.py里还是用到了detail图像，也就是gt图像：
fake_out = fake_out.detach().cpu() * mask + detail*(1-mask)
这里要怎么解释？

**Comment by yzzyqaq** [comment]
![image](https://github.com/KumapowerLIU/Rethinking-Inpainting-MEDFE/assets/102863017/9bcbc65f-fb23-4ce6-9335-1d2aee1fc172)
it succeeds in continue_train
